r/R/expression.R

```diff
@@ -28,8 +28,17 @@
   # stringr spellings of those
   "str_length" = "utf8_length",
   "str_to_lower" = "utf8_lower",
-  "str_to_upper" = "utf8_upper"
+  "str_to_upper" = "utf8_upper",
   # str_trim is defined in dplyr.R
+  "year" = "year",
+  "isoyear" = "iso_year",
+  "quarter" = "quarter",
+  "month" = "month",
+  "day" = "day",
+  "yday" = "day_of_year",
+  "isoweek" = "iso_week",
+  "minute" = "minute",
+  "second" = "second"
```

Comment:
Note that second might do something different. I think "second" in lubridate is the equivalent of "second + subsecond" in arrow

Reply:
Good catch.  Hmm, I think this might be more complicated than that actually, as lubridate rounds it to 1 decimal place, using R's odd-even rounding whereas Arrow doesn't do the rounding, so I'm not sure if this can be done without rounding being implemented (see https://issues.apache.org/jira/browse/ARROW-12744)

Reply:
What do you mean exactly with rounding? A quick try gives me:

```
> second(ymd_hms("2011-06-04 12:00:01.123456"))
[1] 1.123456
```

which seems to give all decimals

Reply:
So it does; I think I must have been getting mixed up with something I'd changed when I needed to update a test I wrote, never mind,

Reply:
What about nanoseconds?
```R
> second(ymd_hms("2011-06-04 12:00:01.123456789"))
[1] 1.123457
```

Arrow would probably return `1.123456789`.

Reply:
I don't think lubridate / R supports nanosecond resolution

Reply:
I think so too. So probably it should be "`second = second + round(subsecond, 6)`" to match that behaviour?

Reply:
IDK that we should truncate the data like this. A slight (technical) API difference is probably better than throwing away precision.

Reply:
Agreed, I would also say that it's not because lubridate does not support nanoseconds that *if* you actually have nanoseconds (which is possible since arrow supports it) those should be discarded.